src/components/ListItem.jsx

```diff
@@ -1,26 +1,46 @@
-import { useEffect, useState } from 'react';
+import { useState, useEffect } from 'react';
+
 import { updateItem } from '../api/firebase';
+import { ONE_DAY_IN_MILLISECONDS } from '../utils';
 
 import './ListItem.css';
 
-export function ListItem({ name, listToken, itemId, dateLastPurchased }) {
-	const [wasPurchased, setWasPurchased] = useState(false);
+export function ListItem({
+	name,
+	listToken,
+	itemId,
+	dateLastPurchased,
+	dateNextPurchased,
+}) {
+	const [prevDateLastPurchased, setPrevDateLastPurchased] = useState(null);
+	const [prevDateNextPurchased, setPrevDateNextPurchased] = useState(null);
+	const [disabled, setDisabled] = useState(false);
 
-	const handleCheck = (checked) => {
-		updateItem(checked, listToken, itemId);
-		setWasPurchased((prevWasPurchased) => !prevWasPurchased);
-	};
+	const currentDate = new Date().getTime();
+	const dateLastPurchasedPlus24h = dateLastPurchased
+		? dateLastPurchased.toDate().getTime() + ONE_DAY_IN_MILLISECONDS
+		: null;
 
+	const wasPurchased = currentDate < dateLastPurchasedPlus24h;
+
+	//disables ability to uncheck item if the item was marked as purchased on page load
 	useEffect(() => {
-		const currentDate = new Date().getTime();
-		const dayInMilliSec = 60 * 60 * 24 * 1000;
+		setDisabled(wasPurchased);
```

Comment:
I noticed that if the user tried to unclick an item that `wasPurchased` (aka clicked) on page load, we would get errors since `prevDateLastPurchased` & `prevDateNextPurchased` would be null since we don't have the data from previous sessions.

So, I was thinking we could disable the checkboxes if they were clicked on page load.
Therefore, if you accidentally click an item as purchased, you can unclick it as long as you are in the same session. This would prevent us from getting any errors.

But, now we have a warning because of the empty dependency array. Wondering if there is a better way to handle this?

Reply:
I'm a little bit stumped on this one 😅
With the current code, is it letting you uncheck the box at all after you check the item?

Reply:
With the current code, the intention was that if the item was already marked as purchased on page load (ie marked as purchased from a previous session), it will be disabled. Therefore, you can only unclick an item in the same session that you accidentally clicked on it.

This was because, in order to unclick the item & maintain its data (info needed for `getFutureDate(estimateOfDate)`), we need to save the items info into state when it's first clicked as purchased.

I tried to save the data for each item when we defined state:
```
	const [prevDateLastPurchased, setPrevDateLastPurchased] = useState(dateLastPurchased);
	const [prevDateNextPurchased, setPrevDateNextPurchased] = useState(DateNextPurchased;
```
But then, the item would always show up as `wasPurchased`, even when unchecked, because `dateLastPurchased` would be within 24hrs, and it started seem like a can of worms 🐛

Reply:
Oh yes, that's tricky. All the state variable interconnectedness makes this a hairy thing to tackle.

Reply:
Super interesting approach here, thanks for the brain workout 😵😄

I was a little unsure about the idea of disabling a checked input like this, but I see what you are going for and I think it works. Maybe we just need to add some sort of accessible labelling down the road so users will understand 'why' its disabled, ya know? It's a cool technical solution, I just worry about the user experience of it.

Undo is just a complicated problem, and think you have a pretty decent little solution here! So well done.

In regards to your useEffect deps issue, I think this captures your intent: 
```
useEffect(() => {
    if (!prevDateLastPurchased && !prevDateNextPurchased)
        setDisabled(wasPurchased);
}, [wasPurchased, prevDateNextPurchased, prevDateLastPurchased]);
```